adminSiteClient/SiteRedirectsIndexPage.tsx

```diff
@@ -0,0 +1,184 @@
+import cx from "classnames"
+import React, { useContext, useEffect, useState } from "react"
+import { useForm } from "react-hook-form"
+
+import { AdminAppContext } from "./AdminAppContext.js"
+import { AdminLayout } from "./AdminLayout.js"
+
+const VALID_PATTERN = /^\/$|^.*[^\/]+$/
```

Comment:
I would be stricter here and have a different pattern for source and target. Sources have to start with a slash. Targets have to start either with a slash or http://.

I think it is also nice to document regexps a bit more since they are so dense - I tried one form of this e.g. [here](https://github.com/owid/owid-grapher/blob/02bd8ba00448254bb89a70bba110371e9dae68a9/packages/%40ourworldindata/components/src/IndicatorKeyData/IndicatorKeyData.tsx#L117)

Reply:
Changed. I made the validation messages more specific, which should also serve as docs for the regexps. Let me know if you think it's not sufficient.